ambry-store/src/test/java/com.github.ambry.store/IndexTest.java

```diff
@@ -1260,6 +1290,89 @@ private void testHardDeletePauseResume(boolean reloadIndex) throws InterruptedEx
     state.verifyEntriesForHardDeletes(idsDeleted);
   }
 
+  // getLogSegmentsNotInJournal helpers
+
+  /**
+   * Tests {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   * @param maxLogSegmentsToBeIgnored number of log segments not to be returned via {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   * @param maxEntriesInJournal max number of entries in {@link Journal}
+   * @param reloadIndex {@code true} if index needs to be reloaded and tested for {@link PersistentIndex#getLogSegmentsNotInJournal()}
+   *                    @@code false} otherwise
+   * @throws InterruptedException
+   * @throws StoreException
+   * @throws IOException
+   */
+  private void testGetLogSegmentsNotInJournal(int maxLogSegmentsToBeIgnored, int maxEntriesInJournal,
```

Comment:
Also, incase we plan to enhance CuratedLogIndexState to work with cleaning up of log and index, adding more tests would be very easy.